numalogic/models/autoencoder/base.py

```diff
@@ -68,8 +85,10 @@ def configure_optimizers(self) -> Dict[str, Any]:
 
     def training_step(self, batch: Tensor, batch_idx: int) -> Tensor:
         loss = self._get_reconstruction_loss(batch)
+        self._total_train_loss += loss.detach().item()
```

Comment:
as you only need a float value, do you still need the detach call? can you just do `loss.item()`?

Reply:
In training, tensors are by default attached to the computational graph (autograd). We need to detach them first, before getting the value.